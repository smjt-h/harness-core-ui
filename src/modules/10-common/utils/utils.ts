/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { IconName } from '@harness/uicore'
import { Scope } from '@common/interfaces/SecretsInterface'
import { ModuleName } from 'framework/types/ModuleName'

interface SetPageNumberProps {
  setPage: (value: React.SetStateAction<number>) => void
  pageItemsCount?: number
  page: number
}

export const getModuleIcon = (module: ModuleName): IconName => {
  switch (module) {
    case ModuleName.CD:
      return 'cd-main'
    case ModuleName.CV:
      return 'cv-main'
    case ModuleName.CI:
      return 'ci-main'
    case ModuleName.CE:
      return 'ce-main'
    case ModuleName.CF:
      return 'cf-main'
  }
  return 'nav-project'
}

export const getReference = (scope?: Scope, identifier?: string): string | undefined => {
  switch (scope) {
    case Scope.PROJECT:
      return identifier
    case Scope.ORG:
      return `org.${identifier}`
    case Scope.ACCOUNT:
      return `account.${identifier}`
  }
}

export const setPageNumber = ({ setPage, pageItemsCount, page }: SetPageNumberProps): void => {
  if (pageItemsCount === 0 && page > 0) {
    setPage(page - 1)
  }
}

export const formatCount = (num: number): string => {
  const min = 1e3
  if (num >= min) {
    const units = ['k', 'M', 'B', 'T']
    const order = Math.floor(Math.log(num) / Math.log(1000))
    const finalNum = Math.round(num / 1000 ** order)
    if (finalNum === min && order < units.length) {
      return 1 + units[order]
    }
    return finalNum + units[order - 1]
  }
  return num.toLocaleString()
}

export const isCommunityPlan = (): boolean => false

export const isOnPrem = (): boolean => window.deploymentType === 'ON_PREM'

export const focusOnNode = (node: HTMLElement): void => {
  const oldTabIndex = node.tabIndex
  node.tabIndex = -1
  node.focus()
  node.tabIndex = oldTabIndex
}

const HOTJAR_SUPPRESSION_ATTR = 'data-hj-suppress'

// Utility to add `data-hj-suppress` into a collection of elements to
// suppress data from HotJar recording
// @see https://bit.ly/3rCgpOY
export const suppressHotJarRecording = (elements: Element[] | null | undefined): void => {
  if (window.hj) {
    elements?.forEach?.((e: Element) => {
      if (!e.hasAttribute(HOTJAR_SUPPRESSION_ATTR)) {
        e.setAttribute(HOTJAR_SUPPRESSION_ATTR, 'true')
      }
    })
  }
}

// Utility to generate { 'data-hj-suppress': true } attribute if HotJar is available
export const addHotJarSuppressionAttribute = (): { [HOTJAR_SUPPRESSION_ATTR]: boolean } | undefined =>
  window.hj ? { [HOTJAR_SUPPRESSION_ATTR]: true } : undefined
