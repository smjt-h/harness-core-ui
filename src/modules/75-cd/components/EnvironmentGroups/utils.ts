/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { defaultTo } from 'lodash-es'
import type {
  EnvironmentGroupFilterProperties,
  EnvironmentGroupResponseDTO,
  GetEnvironmentGroupListQueryParams
} from 'services/cd-ng'

export function cleanData(values: EnvironmentGroupResponseDTO): EnvironmentGroupResponseDTO {
  const {
    name: prevName,
    identifier: prevIdentifier,
    description: prevDescription,
    tags,
    orgIdentifier,
    projectIdentifier,
    envIdentifiers
  } = values

  const name = prevName?.toString().trim()
  const identifier = prevIdentifier?.toString().trim()
  const description = prevDescription?.toString().trim()
  return {
    name,
    identifier,
    description,
    tags: defaultTo(tags, {}),
    orgIdentifier,
    projectIdentifier,
    envIdentifiers
  } as any
}

export enum SortFields {
  LastUpdatedAt = 'lastModifiedAt',
  AZ09 = 'AZ09',
  ZA90 = 'ZA90',
  Name = 'name'
}

export enum Sort {
  DESC = 'DESC',
  ASC = 'ASC'
}

export enum EnvironmentGroupDetailsTab {
  CONFIGURATION = 'CONFIGURATION',
  ENVIRONMENTS = 'ENVIRONMENTS'
}

export interface EnvironmentGroupListQueryParams
  extends Partial<
    Omit<GetEnvironmentGroupListQueryParams, 'sort' | 'getDefaultFromOtherRepo' | 'envGroupIdentifiers'>
  > {
  filters?: EnvironmentGroupFilterProperties
}
