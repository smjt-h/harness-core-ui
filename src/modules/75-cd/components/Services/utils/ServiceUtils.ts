/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

export enum ServiceTabs {
  SUMMARY = 'summaryTab',
  Configuration = 'configuration',
  REFERENCED_BY = 'refrencedByTab',
  ActivityLog = 'activityLog'
}

const DefaultService = {
  serviceDefinition: {
    spec: {
      variables: []
    }
  }
}

export const DefaultNewStageName = 'Stage Name'
export const DefaultNewStageId = 'stage_id'

export const initialServiceState = {
  service: { ...DefaultService }
  // isLoading: false,
  // isBETemplateUpdated: false,
  // isDBInitialized: false,
  // isUpdated: false,
  // isInitialized: false,
  // gitDetails: {},
  // entityValidityDetails: {}
}
