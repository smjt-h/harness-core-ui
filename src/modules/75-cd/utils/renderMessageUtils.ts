/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { BannerType } from '@common/layouts/Constants'
import {
  getActiveUsageNumber,
  isFeatureLimitBreached,
  isFeatureOveruseActive,
  isFeatureWarningActive
} from '@common/layouts/FeatureBanner'
import type { CheckFeatureReturn } from 'framework/featureStore/featureStoreUtil'
import type { UseStringsReturn } from 'framework/strings'

interface BannerText {
  message: () => string
  bannerType: BannerType
}

export const getBannerText = (
  getString: UseStringsReturn['getString'],
  additionalLicenseProps: Record<string, unknown>,
  serviceFeatureDetail?: CheckFeatureReturn,
  dpmFeatureDetail?: CheckFeatureReturn,
  initialDeploymentsFeatureDetail?: CheckFeatureReturn
): BannerText => {
  const { isFreeEdition: isCDFree, isTeamEdition: isCDTeam } = additionalLicenseProps
  // Check for limit breach
  const isServiceLimitBreached = isFeatureLimitBreached(serviceFeatureDetail)
  const isDpmLimitBreached = isFeatureLimitBreached(dpmFeatureDetail)
  let limitBreachMessageString = ''

  if (isServiceLimitBreached && isDpmLimitBreached) {
    limitBreachMessageString = getString('cd.featureRestriction.banners.serviceAndDeploymentsLevelUp', {
      deploymentsLimit: dpmFeatureDetail?.featureDetail?.limit,
      serviceLimit: serviceFeatureDetail?.featureDetail?.limit
    })
  } else if (isServiceLimitBreached && isCDFree) {
    limitBreachMessageString = getString('cd.featureRestriction.banners.serviceLevelUp', {
      serviceLimit: serviceFeatureDetail?.featureDetail?.limit
    })
  } else if (isServiceLimitBreached && isCDTeam) {
    limitBreachMessageString = getString('cd.featureRestriction.banners.serviceLevelUpTeamEnterprise', {
      serviceLimit: serviceFeatureDetail?.featureDetail?.limit
    })
  } else if (isDpmLimitBreached) {
    limitBreachMessageString = getString('cd.featureRestriction.banners.deploymentsPerMonthLevelUp', {
      count: dpmFeatureDetail?.featureDetail?.count,
      deploymentsLimit: dpmFeatureDetail?.featureDetail?.limit
    })
  }

  if (limitBreachMessageString) {
    return {
      message: () => limitBreachMessageString,
      bannerType: BannerType.LEVEL_UP
    }
  }

  // Checking for limit usage warning
  let warningMessageString = ''
  const isServiceWarningActive = isFeatureWarningActive(serviceFeatureDetail)
  const isDpmWarningActive = isFeatureWarningActive(dpmFeatureDetail)
  const isInitialDeplWarningActive = isFeatureWarningActive(initialDeploymentsFeatureDetail)
  if (isInitialDeplWarningActive) {
    warningMessageString = getString('cd.featureRestriction.banners.initialDeploymentsWarningActive', {
      warningLimit: getActiveUsageNumber(initialDeploymentsFeatureDetail)
    })
  } else if (isServiceWarningActive) {
    warningMessageString = getString('cd.featureRestriction.banners.serviceWarningActive', {
      warningLimit: getActiveUsageNumber(serviceFeatureDetail)
    })
  } else if (isDpmWarningActive) {
    warningMessageString = getString('cd.featureRestriction.banners.dpmWarningActive', {
      count: dpmFeatureDetail?.featureDetail?.count,
      warningLimit: dpmFeatureDetail?.featureDetail?.limit
    })
  }

  if (warningMessageString) {
    return {
      message: () => warningMessageString,
      bannerType: BannerType.INFO
    }
  }

  let overuseMessageString = ''
  const isServiceOveruseActive = isFeatureOveruseActive(serviceFeatureDetail)
  if (isServiceOveruseActive && isCDTeam) {
    overuseMessageString = getString('cd.featureRestriction.banners.serviceOveruseTeamEnterprise')
  }
  if (overuseMessageString) {
    return {
      message: () => overuseMessageString,
      bannerType: BannerType.OVERUSE
    }
  }

  // If neither of limit breach/ warning/ overuse needs to be shown, return with an empty string.
  // This will ensure no banner is shown
  return {
    message: () => '',
    bannerType: BannerType.LEVEL_UP
  }
}
