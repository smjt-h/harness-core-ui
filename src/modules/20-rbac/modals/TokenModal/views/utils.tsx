/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { getReadableDateTime } from '@common/utils/dateUtils'

export const DATE_FORMAT = 'MM/DD/YYYY'

export const hasNoExpiryDate = (value?: number): boolean => {
  return getReadableDateTime(value, DATE_FORMAT) === '12/31/2099'
}

export const getSelectedExpiration = (value?: number): string => {
  if (!value) {
    return '30'
  } else if (hasNoExpiryDate(value)) {
    return '-1'
  }
  return 'custom'
}

export const getSelectedExpirationDate = (value: string): string => {
  switch (value) {
    case '30':
    case '90':
    case '180':
      return getReadableDateTime(new Date().setDate(new Date().getDate() + parseInt(value)), DATE_FORMAT)
    case '-1':
      return '12/31/2099'
    default:
      return ''
  }
}

export const getExpirationDate = (value?: number): string => {
  if (value) {
    return getReadableDateTime(value, DATE_FORMAT)
  } else {
    return getSelectedExpirationDate(getSelectedExpiration(value))
  }
}
