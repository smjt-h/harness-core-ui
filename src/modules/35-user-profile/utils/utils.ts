/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { IconName } from '@wings-software/uicore'
import { ValueType } from '@secrets/components/TextReference/TextReference'
import type { SCMData } from '@user-profile/modals/SourceCodeManager/views/SourceCodeManagerForm'
import type { SourceCodeManagerAuthentication, SourceCodeManagerDTO } from 'services/cd-ng'
import { setSecretField } from '@secrets/utils/SecretField'

export enum ConnectionType {
  HTTPS = 'HTTPS',
  HTTP = 'Http',
  SSH = 'Ssh'
}

export enum AuthTypes {
  USERNAME_PASSWORD = 'UsernamePassword',
  USERNAME_TOKEN = 'UsernameToken',
  KERBEROS = 'Kerberos',
  SSH_KEY = 'SSH_KEY',
  AWSCredentials = 'AWSCredentials'
}

export interface SourceCodeType {
  text: string
  value: SourceCodeTypes
  icon: IconName
}

export enum SourceCodeTypes {
  BITBUCKET = 'BITBUCKET',
  GITHUB = 'GITHUB',
  GITLAB = 'GITLAB',
  AZURE_REPO = 'AZURE_REPO',
  AWS_CODE_COMMIT = 'AWS_CODE_COMMIT'
}

export const selectedValueToTypeMap: Record<SourceCodeTypes, SourceCodeTypes> = {
  [SourceCodeTypes.BITBUCKET]: SourceCodeTypes.BITBUCKET,
  [SourceCodeTypes.GITHUB]: SourceCodeTypes.GITHUB,
  [SourceCodeTypes.GITLAB]: SourceCodeTypes.GITLAB,
  [SourceCodeTypes.AWS_CODE_COMMIT]: SourceCodeTypes.AWS_CODE_COMMIT,
  [SourceCodeTypes.AZURE_REPO]: SourceCodeTypes.AZURE_REPO
}

export const getIconBySCM = (item: SourceCodeTypes): IconName => {
  switch (item) {
    case SourceCodeTypes.BITBUCKET:
      return 'bitbucket-blue'
    case SourceCodeTypes.GITHUB:
      return 'github'
    case SourceCodeTypes.GITLAB:
      return 'service-gotlab'
    case SourceCodeTypes.AWS_CODE_COMMIT:
      return 'service-aws-code-deploy'
    case SourceCodeTypes.AZURE_REPO:
      return 'service-azure'
    default:
      return 'bitbucket'
  }
}

export const getAuthentication = (values: SCMData): SourceCodeManagerAuthentication | undefined => {
  switch (values.authType) {
    case AuthTypes.USERNAME_PASSWORD:
      return {
        type: ConnectionType.HTTP,
        spec: {
          type: AuthTypes.USERNAME_PASSWORD,
          spec: {
            ...(values.username?.type === ValueType.TEXT
              ? { username: values.username.value }
              : { usernameRef: values.username?.value }),
            passwordRef: values.password?.referenceString
          }
        }
      }
    case AuthTypes.USERNAME_TOKEN:
      return {
        type: ConnectionType.HTTP,
        spec: {
          type: AuthTypes.USERNAME_TOKEN,
          spec: {
            ...(values.username?.type === ValueType.TEXT
              ? { username: values.username.value }
              : { usernameRef: values.username?.value }),
            tokenRef: values.accessToken?.referenceString
          }
        }
      }
    default:
      return undefined
  }
}

const getBitBucketFormData = async (
  sourceCodeManagerData: SourceCodeManagerDTO,
  accountIdentifier: string
): Promise<SCMData> => {
  const { name, authentication } = sourceCodeManagerData
  return {
    name: name,
    authType: authentication?.spec?.type,
    password: (await setSecretField(authentication?.spec?.spec?.passwordRef, {
      accountIdentifier
    })) as SCMData['password'],
    username: {
      value: authentication?.spec?.spec?.username || authentication?.spec?.spec?.usernameRef,
      type: authentication?.spec?.spec?.usernameRef ? ValueType.ENCRYPTED : ValueType.TEXT
    }
  }
}

const getGithubFormData = async (
  sourceCodeManagerData: SourceCodeManagerDTO,
  accountIdentifier: string
): Promise<SCMData> => {
  const { name, authentication } = sourceCodeManagerData
  return {
    name: name,
    authType: authentication?.spec?.type,
    accessToken: (await setSecretField(
      authentication?.spec?.spec?.tokenRef || authentication?.spec?.apiAccess?.spec?.tokenRef,
      { accountIdentifier }
    )) as SCMData['accessToken'],
    username: {
      value: authentication?.spec?.spec?.username || authentication?.spec?.spec?.usernameRef,
      type: authentication?.spec?.spec?.usernameRef ? ValueType.ENCRYPTED : ValueType.TEXT
    }
  }
}

export const getFormDataBasedOnSCMType = (
  sourceCodeManagerData: SourceCodeManagerDTO,
  accountIdentifier: string
): Promise<SCMData | undefined> => {
  switch (sourceCodeManagerData.type) {
    case SourceCodeTypes.BITBUCKET:
      return getBitBucketFormData(sourceCodeManagerData, accountIdentifier)
    case SourceCodeTypes.GITHUB:
      return getGithubFormData(sourceCodeManagerData, accountIdentifier)
    default:
      return Promise.resolve(undefined)
  }
}

export const getDefaultSCMType = (
  supportedSCMS: SourceCodeType[],
  currentSCM?: SourceCodeManagerDTO['type']
): SourceCodeType => {
  return supportedSCMS.find(item => item.value === currentSCM) || supportedSCMS[0]
}

export const getDefaultSelected = (type?: SourceCodeTypes): AuthTypes | undefined => {
  switch (type) {
    case SourceCodeTypes.GITHUB:
      return AuthTypes.USERNAME_TOKEN
    case SourceCodeTypes.BITBUCKET:
    case SourceCodeTypes.GITLAB:
    case SourceCodeTypes.AZURE_REPO:
      return AuthTypes.USERNAME_PASSWORD
    case SourceCodeTypes.AWS_CODE_COMMIT:
      return AuthTypes.AWSCredentials
    default:
      return undefined
  }
}
