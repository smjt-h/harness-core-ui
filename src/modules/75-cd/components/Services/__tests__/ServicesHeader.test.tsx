/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, fireEvent, getByText } from '@testing-library/react'
import { TestWrapper, findDialogContainer } from '@common/utils/testUtils'
import { ServicesHeader } from '@cd/components/Services/ServicesHeader/ServicesHeader'

jest.mock('services/cd-ng', () => {
  return {
    useGetYamlSchema: jest.fn(() => ({ data: null })),
    useDeleteServiceV2: jest.fn(() => ({ mutate: jest.fn() })),
    useCreateServicesV2: jest.fn(() => ({ data: null })),
    useUpsertServiceV2: jest.fn(() => ({ data: null }))
  }
})

describe('ServiceHeader', () => {
  test('render', () => {
    const { container } = render(
      <TestWrapper
        path="account/:accountId/cd/orgs/:orgIdentifier/projects/:projectIdentifier/services"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <ServicesHeader />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
  test('should open and close the Services modal', async () => {
    const { container } = render(
      <TestWrapper>
        <ServicesHeader />
      </TestWrapper>
    )
    fireEvent.click(container.querySelector('[data-testid="add-service"]') as HTMLElement)

    const dialog = findDialogContainer() as HTMLElement
    expect(dialog).toMatchSnapshot()
    expect(dialog).toBeTruthy()
    expect(getByText(document.body, 'cancel')).toBeDefined()
    fireEvent.click(getByText(document.body, 'cancel') as HTMLButtonElement)
    expect(findDialogContainer()).toBeFalsy()
  })
})
