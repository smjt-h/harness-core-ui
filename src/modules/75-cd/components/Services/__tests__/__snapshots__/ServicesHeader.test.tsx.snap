// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceHeader render 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main header StyledProps--flex StyledProps--flex-distribution-space-between"
  >
    <div
      class="Layout--horizontal StyledProps--main"
    >
      <button
        class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color Button--withLeftIcon"
        data-testid="add-service"
        type="button"
      >
        <span
          class="bp3-icon bp3-icon-plus StyledProps--main StyledProps--padding-right-xsmall"
          icon="plus"
        >
          <svg
            data-icon="plus"
            height="10"
            viewBox="0 0 16 16"
            width="10"
          >
            <desc>
              plus
            </desc>
            <path
              d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span
          class="bp3-button-text"
        >
          newService
        </span>
      </button>
    </div>
    <div
      class="Layout--horizontal StyledProps--main StyledProps--flex"
    >
      <button
        class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color Button--without-shadow Button--withLeftIcon"
        type="button"
      >
        <span
          class="bp3-icon StyledProps--main"
          data-icon="insight-view"
        >
          <svg
            height="16"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="6"
              rx="1"
              width="19"
              x="2"
              y="4"
            />
            <rect
              height="2"
              rx="1"
              width="19"
              x="2"
              y="13"
            />
            <rect
              height="2"
              rx="1"
              width="19"
              x="2"
              y="18"
            />
          </svg>
        </span>
      </button>
      <button
        class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--without-shadow Button--withLeftIcon"
        type="button"
      >
        <span
          class="bp3-icon StyledProps--main"
          data-icon="list-view"
        >
          <svg
            height="16"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="2"
              rx="1"
              width="19"
              x="2"
              y="4"
            />
            <rect
              height="2"
              rx="1"
              width="19"
              x="2"
              y="11"
            />
            <rect
              height="2"
              rx="1"
              width="19"
              x="2"
              y="18"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`ServiceHeader should open and close the Services modal 1`] = `
<div
  class="bp3-dialog padded-dialog dialogStyles"
>
  <div
    class="bp3-dialog-header"
  >
    <h4
      class="bp3-heading"
    >
      newService
    </h4>
    <button
      aria-label="Close"
      class="bp3-button bp3-minimal bp3-dialog-close-button"
      type="button"
    >
      <span
        class="bp3-icon bp3-icon-small-cross"
        icon="small-cross"
      >
        <svg
          data-icon="small-cross"
          height="20"
          viewBox="0 0 20 20"
          width="20"
        >
          <desc>
            small-cross
          </desc>
          <path
            d="M11.41 10l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L10 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42L8.59 10 5.3 13.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l3.29-3.3 3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L11.41 10z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </button>
  </div>
  <div
    class="StyledProps--font StyledProps--main yamlToggle"
  >
    <div
      class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-flex-start StyledProps--padding-top-small"
    >
      <div
        class="PillToggle--optionBtns"
      >
        <div
          class="PillToggle--item PillToggle--selected"
          data-name="toggle-option-one"
          role="button"
          tabindex="0"
        >
          VISUAL
        </div>
        <div
          class="PillToggle--item"
          data-name="toggle-option-two"
          role="button"
          tabindex="0"
        >
          YAML
        </div>
      </div>
    </div>
  </div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <form
      action="#"
      class="FormikForm--main"
    >
      <div>
        <div
          class="StyledProps--font StyledProps--main main"
        >
          <div
            class="InputWithIdentifier--txtNameContainer TextInput--main"
          >
            <div
              class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main InputWithIdentifier--txtIdContainer"
            >
              <div
                class="Text--withDocsTooltip"
                data-tooltip-id="deployService_identifier"
              >
                <p
                  class="StyledProps--font StyledProps--main"
                >
                  Id
                </p>
              </div>
              <p
                class="StyledProps--font StyledProps--main"
              >
                :
              </p>
              <span
                class="bp3-popover-wrapper"
              >
                <span
                  class="bp3-popover-target"
                >
                  <div
                    class="bp3-editable-text bp3-editable-text-placeholder InputWithIdentifier--idValue"
                    tabindex="0"
                  >
                    <span
                      class="bp3-editable-text-content"
                      style="height: 1px; line-height: 1px; min-width: 0;"
                    />
                  </div>
                </span>
              </span>
              <span
                class="bp3-icon StyledProps--main"
                data-icon="Edit"
                style="vertical-align: middle; cursor: pointer;"
              >
                <svg
                  height="12"
                  viewBox="0 0 16 16"
                  width="12"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M11.207.485a.5.5 0 01.354-.146 3.5 3.5 0 013.5 3.5.5.5 0 01-.146.353l-11 11a.5.5 0 01-.354.147h-3a.5.5 0 01-.5-.5v-3a.5.5 0 01.146-.354l11-11zm.553.862L1.061 12.046v2.293h2.293l10.7-10.7a2.5 2.5 0 00-2.294-2.292z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M11.89 2.646a.5.5 0 010 .708L8.353 6.889a.5.5 0 01-.708-.707l3.536-3.536a.5.5 0 01.707 0z"
                    fill="currentColor"
                    fill-rule="evenodd"
                    opacity="0.3"
                  />
                </svg>
              </span>
            </div>
            <div
              class="bp3-form-group"
              data-id="name-0"
            >
              <label
                class="bp3-label"
                for="name"
              >
                <span
                  class="Tooltip--acenter"
                  data-tooltip-id="deployService_name"
                >
                  name
                </span>
                 
                <span
                  class="bp3-text-muted"
                />
              </label>
              <div
                class="bp3-form-content"
              >
                <div
                  class="bp3-input-group"
                >
                  <input
                    autocomplete="off"
                    class="bp3-input"
                    name="name"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="StyledProps--font StyledProps--main"
          >
            <label
              class="Label--main bp3-label descriptionLabel"
            >
              optionalField
              <span
                class="bp3-icon StyledProps--main editOpen"
                data-icon="Edit"
                data-name="edit"
                data-testid="description-edit"
              >
                <svg
                  height="12"
                  viewBox="0 0 16 16"
                  width="12"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M11.207.485a.5.5 0 01.354-.146 3.5 3.5 0 013.5 3.5.5.5 0 01-.146.353l-11 11a.5.5 0 01-.354.147h-3a.5.5 0 01-.5-.5v-3a.5.5 0 01.146-.354l11-11zm.553.862L1.061 12.046v2.293h2.293l10.7-10.7a2.5 2.5 0 00-2.294-2.292z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M11.89 2.646a.5.5 0 010 .708L8.353 6.889a.5.5 0 01-.708-.707l3.536-3.536a.5.5 0 01.707 0z"
                    fill="currentColor"
                    fill-rule="evenodd"
                    opacity="0.3"
                  />
                </svg>
              </span>
            </label>
          </div>
          <div
            class="StyledProps--font StyledProps--main"
          >
            <label
              class="Label--main bp3-label descriptionLabel"
            >
              optionalField
              <span
                class="bp3-icon StyledProps--main editOpen"
                data-icon="Edit"
                data-name="edit"
                data-testid="tags-edit"
              >
                <svg
                  height="12"
                  viewBox="0 0 16 16"
                  width="12"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M11.207.485a.5.5 0 01.354-.146 3.5 3.5 0 013.5 3.5.5.5 0 01-.146.353l-11 11a.5.5 0 01-.354.147h-3a.5.5 0 01-.5-.5v-3a.5.5 0 01.146-.354l11-11zm.553.862L1.061 12.046v2.293h2.293l10.7-10.7a2.5 2.5 0 00-2.294-2.292z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M11.89 2.646a.5.5 0 010 .708L8.353 6.889a.5.5 0 01-.708-.707l3.536-3.536a.5.5 0 01.707 0z"
                    fill="currentColor"
                    fill-rule="evenodd"
                    opacity="0.3"
                  />
                </svg>
              </span>
            </label>
          </div>
        </div>
        <div
          class="Layout--horizontal Layout--layout-spacing-small StyledProps--main StyledProps--padding-top-xlarge"
        >
          <button
            class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--variation Button--variation-primary"
            data-id="service-save"
            type="submit"
          >
            <span
              class="bp3-button-text"
            >
              save
            </span>
          </button>
          <button
            class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--variation Button--variation-tertiary"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              cancel
            </span>
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
